mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,8 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+        require(indices.size == 0 || indices(0) >= 0,
```

Comment:
Thanks for review . I'm thinking it has already been converted to zero based on line 81.